cristin-commons/src/main/java/no/unit/nva/cristin/model/Constants.java

```diff
@@ -67,9 +82,153 @@ public class Constants {
+        PATH_PROJECT("projects", "project", PATTERN_IS_NUMBER6),
+        BIOBANK("biobank"),
+        FUNDING("funding"),
+        FUNDING_SOURCE("funding_source"),
+        GRANT_ID("project_code", null, PATTERN_IS_NUMBER6),
+        INSTITUTION("institution"),
+        LANGUAGE("lang", "language", PATTERN_IS_LANGUAGE),
+        LEVELS("levels", "depth", PATTERN_IS_NUMBER6),
+        NAME("name"),
+        PROJECT_APPROVAL_REFERENCE_ID("approval_reference_id"),
+        PROJECT_APPROVED_BY("approved_by"),
+        PROJECT_KEYWORD("keyword"),
+        PROJECT_ORGANIZATION("parent_unit_id","organization", PATTERN_IS_URL,
+            INVALID_URI_MESSAGE,true),
+        PROJECT_MANAGER("project_manager"),
+        PROJECT_MODIFIED_SINCE("modified_since", null, PATTERN_IS_DATE),
+        PROJECT_PARTICIPANT("participant"),
+        PROJECT_UNIT("unit"),
+        QUERY("query", null, PATTERN_IS_NON_EMPTY,
+              invalidQueryParametersMessage("query", ALPHANUMERIC_CHARACTERS_DASH_COMMA_PERIOD_AND_WHITESPACE),
+              true),
+        STATUS("status", PATTERN_IS_STATUS, true),
+        TITLE("title", null, PATTERN_IS_TITLE,
+            invalidQueryParametersMessage("title", ALPHANUMERIC_CHARACTERS_DASH_COMMA_PERIOD_AND_WHITESPACE),
+            true),
+        USER("user"),
+        PAGE_CURRENT("page", null, PATTERN_IS_NUMBER6,
+            ERROR_MESSAGE_INVALID_NUMBER, false),
+        PAGE_ITEMS_PER_PAGE("per_page", "results",
+            PATTERN_IS_NUMBER6, ERROR_MESSAGE_INVALID_NUMBER, false),
+        PAGE_SORT("sort");
+
+        public final static int IGNORE_PATH_PARAMETER_INDEX = 3;
+
+        public static final Set<QueryParameterKey> VALID_QUERY_PARAMETERS =
+            Arrays.stream(QueryParameterKey.values())
+                .filter(f -> f.ordinal() > IGNORE_PATH_PARAMETER_INDEX)
+                .collect(Collectors.toSet());
+
+        public static final Set<String> VALID_QUERY_PARAMETER_KEYS =
+            VALID_QUERY_PARAMETERS.stream()
+                .sorted()
+                .map(QueryParameterKey::getKey)
+                .collect(Collectors.toSet());
+
+        public static final Set<String> VALID_QUERY_PARAMETER_NVA_KEYS =
+            VALID_QUERY_PARAMETERS.stream()
+                .sorted()
+                .map(QueryParameterKey::getNvaKey)
+                .collect(Collectors.toSet());
+
+        private final String pattern;
+        private final String cristinKey;
+        private final String nvaKey;
+        private final boolean encode;
+        private final String errorMessage;
+
+        QueryParameterKey(String cristinKey) {
+            this(cristinKey, null, PATTERN_IS_NON_EMPTY, null, false);
+        }
+
+        QueryParameterKey(String cristinKey, String pattern, boolean encode) {
+            this(cristinKey, null, pattern, null, encode);
+        }
+
+        QueryParameterKey(String cristinKey, String nvaKey, String pattern) {
```

Comment:
It can be confusing on overloading the constructor when you have two constructors with 3 parameters where the only difference on the 3rd param is the type, boolean vs String. Is it not better having them in order and then just passing null or false when not supplied, since all overloads call the largest constructor?